lib/connection_pool/timed_stack.rb

```diff
@@ -93,6 +95,7 @@ def shutdown(&block)
 
       shutdown_connections
     end
+    @shutdown_block = nil if reload
```

Comment:
Should this line go inside the mutex sync block? Otherwise there appears to be a race with pop.

Reply:
You’re correct, if `pop` L70 is called in another thread between L96 & L98 then `pop` will incorrectly raise PoolShuttingDownError. I’ve amended the commit to move this inside the synchronize.